sdk/python/kfp/components/structures.py

```diff
@@ -574,10 +574,11 @@ def load_from_component_yaml(cls, component_yaml: str) -> 'ComponentSpec':
         Returns:
             Component spec in the form of V2 ComponentSpec.
         """
+
         json_component = yaml.safe_load(component_yaml)
         try:
             return ComponentSpec.parse_obj(json_component)
-        except pydantic.ValidationError:
+        except (pydantic.ValidationError, AttributeError):
```

Comment:
@ji-yaqi, I opted to broaden the range of exceptions that will be caught, but still left it more narrow than `except Exception:`. Can you let me know what you think?

Reply:
Yea we could leave it here but I would suspect to have more than attributeError here. I can't think of any for now, and we can add more in the future if needed.